test/hotspot/jtreg/compiler/c2/TestShiftRightAndAccumulate.java

```diff
@@ -25,8 +25,17 @@
  * @test
  * @bug 8260585
  * @summary AArch64: Wrong code generated for shifting right and accumulating four unsigned short integers.
+ *
  * @run main/othervm compiler.c2.TestShiftRightAndAccumulate
  * @run main/othervm -Xcomp compiler.c2.TestShiftRightAndAccumulate
+ */
+
+/**
+ * @test
+ * @bug 8260585
+ * @summary AArch64: Wrong code generated for shifting right and accumulating four unsigned short integers.
+ * @requires vm.compiler2.enabled
+ *
```

Comment:
Why does this one have a duplicate header?

Reply:
Originally, the test case included 3 `@run`s, one without any arguments, the second with `-Xcomp` and the third with `-XX:-SuperWordLoopUnrollAnalysis`. Since the third one is C2 specific, I split up the test case into two separated `@test` segments.
The first one, without any `@requires` includes the first two `@run`s, which are always valid. The second `@test` includes the `@run` with the C2 specific flag and has an `@requires vm.compiler2.enabled`.

Reply:
Yes, jtreg supports it. We have already such cases:
https://github.com/openjdk/jdk/blob/master/test/hotspot/jtreg/compiler/c1/CanonicalizeGetModifiers.java
Jtreg will execute them as separate tests with `#id0` and `#id1` suffixes to test name.

Reply:
Ok, good to know!